File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L4/06/arraylebihdarisatu.py
import numpy as np

print("Enter a 4 by 4 matrix row by row: ")
matrix = np.zeros((4, 4))

for x in range(4):
    for y in range(4):
        matrix[x][y] = float(input())

hasil = np.trace(matrix)

print(f"Sum of the elements in the major diagonal is {hasil}")

File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L4/06/arraylebihdarisatu.kt
package IR

import java.util.*

object arraylebihdarisatu {
    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)


        print("Enter a 4 by 4 matrix row by row: ")
        val matrix = Array(4) { DoubleArray(4) } //inisialisai array 2 dimensi

        for (x in 0..3) {
            for (y in 0..3) {
                matrix[x][y] = sc.nextDouble() //menginput angka ke dalam array 2d
            }
        }
        var hasil = 0.0
        for (i in matrix.indices) {
            hasil += matrix[i][i]
        }

        print("Sum of the elements in the major  diagonal is  $hasil") //menampilkan fungsi hasil perhitungan diagonal
    }
}
